### PR TITLE
PE-45 | Extracted Auth to Own Services File

### PR DESCRIPTION
**Pull request by tonynguyen98**
**What I Did:**
- Extracted any Firebase functions into it's own services file.

**How to Test:**
- Make sure that the login functionality is not broken.

### CODE REVIEW

**Review by txt020 — APPROVED** [review]
Everything looks good!
Login in function works and I was able to sign out.

**Review by dominikyz — CHANGES_REQUESTED** [review]
Was able to sign up with email. Was able to log in with email and was able to log in with google account.
Please add " space character" catcher. 
If email contains space, app doesn't accept email

**Comment by dominikyz** [comment]
Everything works well now.
Great job!

**Review by dominikyz — APPROVED** [review]
Code looks well organized.
Everything works great now. 
Good job!

File: lib/screens/login_page.dart [hunk 1]
```diff
@@ -15,18 +15,22 @@ class LoginPage extends StatefulWidget {
...
+  EmailAuth _emailAuth = new EmailAuth();
+  GoogleAuth _googleAuth = new GoogleAuth();
+  FirebaseExceptionHandler _firebaseExceptionHandler =
+      new FirebaseExceptionHandler();
+
   String email;
   String password;
   bool showSpinner = false;
   String exception = "";
+
```
Comment (ayeejane) [inline lib/screens/login_page.dart:28]: Regex is not working properly

**Review by ayeejane — CHANGES_REQUESTED** [review]

**Review by ayeejane — APPROVED** [review]
Good fix!

**Pull request merged**